neat_cache/lib/neat_cache.dart

```diff
@@ -175,23 +175,31 @@ class _Entry<T, V> implements Entry<T> {
       final created = await create();
       if (created != null) {
         // Calling `set(null)` is equivalent to `purge()`, we can skip that here
-        await set(created, ttl);
+        if (ttl != null) {
+          await set(created, ttl);
+        } else {
+          await set(created);
+        }
       }
       return created;
     }
     return _owner._codec.decode(value);
   }
 
   @override
-  Future<T> set(T value, [Duration ttl]) async {
+  Future<T?> set(T? value, [Duration? ttl]) async {
     if (value == null) {
       await purge();
       return null;
     }
     ttl ??= _owner._ttl;
     final raw = _owner._codec.encode(value);
     try {
-      await _owner._provider.set(_key, raw, ttl);
+      if (ttl != null) {
+        await _owner._provider.set(_key, raw, ttl);
+      } else {
+        await _owner._provider.set(_key, raw);
```

Comment:
why can't we use `set(_key, raw, ttl)` unconditionally?

Reply:
If I allow `Duration?` in the `CacheProvider` abstraction... then we don't need the conditional here... and then I think we'll all be reasonably happy :D